msal/application.py

```diff
@@ -260,6 +301,27 @@ def _build_telemetry_context(
             self._telemetry_buffer, self._telemetry_lock, api_id,
             correlation_id=correlation_id, refresh_reason=refresh_reason)
 
+    def _get_regional_authority(self, central_authority):
+        is_region_specified = bool(self._region_configured
+            and self._region_configured != self.ATTEMPT_REGION_DISCOVERY)
+        self._region_detected = self._region_detected or _detect_region(
+            self.http_client if self._region_configured is not None else None)
+        if (is_region_specified and self._region_configured != self._region_detected):
+            logger.warning('Region configured ({}) != region detected ({})'.format(
+                repr(self._region_configured), repr(self._region_detected)))
+        region_to_use = (
+            self._region_configured if is_region_specified else self._region_detected)
```

Comment:
In case _region_specified is different from _region_detected what will be the value of region_to_use here?

Reply:
It will use `_region_configured`.
PS: That "... if ... else ..." is the Python way of ternary operator. It is supposed to mimic the normal English "outcome = advanced if won else eliminated".